ONSdigital/rm-action-service
# Add action plan selectors

# Motivation and Context
Action plan selectors are our new solution for determining which action plan a case should be assigned to. These selectors are defined by a general jsonb field which for business surveys are populated by `exerciseRef, surveyRef, activeEnrolment` fields but these could be different for different kinds of surveys.

# What has changed
`actionplanselector` column created with liquibase script
Allow `selectors` field to be passed into POST and PUT requests to `/actionplans(/{id})` which updates the new table
Allow general selectors to be passed in as query string parameters to GET `/actionplans` and retrieve plans based on these selectors
Python script added to the `scripts` folder which creates action plan selectors from existing casetypeoverrides in collection exercise service

# How to test?
`mvn clean install` on the associated [actionsvc-api](https://github.com/ONSdigital/rm-actionsvc-api/tree/add-action-plan-selectors) branch
Try a POST request to `/actionplans` with the selectors field
Try a PUT request to `/actionplans/{id}` with the selectors field for an existing action plan
Try to retrieve an action plan by it's selectors with GET `/actionplans?selector_key=selector_value`
`create-action-plan-selectors` script should create the correct action plan selectors based on the existing casetypeoverrides
Unit/Acceptance tests should pass

# Links
[Trello](https://trello.com/c/piwT43rb/9-create-actionplan-selectors-action)
[actionsvc-api PR](https://github.com/ONSdigital/rm-actionsvc-api/pull/15)
[actionsvc-api PR #2](https://github.com/ONSdigital/rm-actionsvc-api/pull/19)
